ovos_solver_openai_persona/rag.py

```
@@ -0,0 +1,500 @@
+            data = response.json()
+
+            if "data" not in data or not isinstance(data["data"], list):
+                raise RequestException(f"Unexpected response format from RAG search: {data}")
+
+            # Extract content from search results
+            retrieved_chunks = [item["content"] for item in data["data"] if "content" in item]
+            LOG.debug(f"Retrieved {len(retrieved_chunks)} chunks from vector store.")
+            return retrieved_chunks
+        except requests.exceptions.RequestException as e:
+            LOG.error(f"Error during RAG search request to {search_url}: {e}")
+            raise RequestException(f"Failed to retrieve context from vector store: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)